src/sagemaker/jumpstart/types.py

```diff
@@ -736,6 +736,334 @@ def _get_regional_property(
+        """
+        self.from_json(spec)
+
+    def from_json(self, json_obj: Dict[str, Any]) -> None:
+        """Sets fields in object based on json.
+
+        Args:
+            json_obj (Dict[str, Any]): Dictionary representation of training config components.
+        """
+        if json_obj is None:
+            return
+
+        self.component_name: str = json_obj["component_name"]
+        self.supported_training_instance_types: Optional[List[str]] = json_obj.get(
+            "supported_training_instance_types"
+        )
+        self.training_artifact_key: Optional[str] = json_obj.get("training_artifact_key")
+        self.training_instance_type_variants: Optional[JumpStartInstanceTypeVariants] = (
+            JumpStartInstanceTypeVariants(json_obj["training_instance_type_variants"])
+            if json_obj.get("training_instance_type_variants")
+            else None
+        )
+        self.tags: Optional[List[str]] = json_obj.get("tags")
+        self.supported_incremental_training_configs: Optional[List[str]] = json_obj.get(
+            "supported_incremental_training_configs"
+        )
+        self.supported_inference_configs: Optional[List[str]] = json_obj.get(
+            "supported_inference_configs"
+        )
+
+    def to_json(self) -> Dict[str, Any]:
+        """Returns json representation of JumpStartTrainingComponent object."""
+        json_obj = {att: getattr(self, att) for att in self.__slots__ if hasattr(self, att)}
+        return json_obj
+
+
+class JumpStartAlternativeConfig(JumpStartDataHolderType):
+    """Data class JumpStart alternative config."""
+
+    __slots__ = ["config_name", "component_names", "components"]
+
+    def __init__(
+        self,
+        spec: Dict[str, Any],
+        components: Dict[str, Union[JumpStartInferenceComponent, JumpStartTrainingComponent]],
+    ):
+        """Initializes a JumpStartAlternativeConfig object
+
+        Args:
+            spec (Dict[str, Any]): Dictionary representation of alternative config.
+            inference_components (Dict[str, Any]): Config components that are used to constuct the final config.
+        """
+        self.from_json(spec)
+        self.components: Dict[
+            str, Union[JumpStartInferenceComponent, JumpStartTrainingComponent]
+        ] = components
+
+    def from_json(self, json_obj: Dict[str, Any]) -> None:
+        """Sets fields in object based on json.
+
+        Args:
+            json_obj (Dict[str, Any]): Dictionary representation of alternative config.
+        """
+
+        self.config_name = json_obj["config_name"]
+        self.component_names = json_obj["component_names"]
+
+    def to_json(self) -> Dict[str, Any]:
+        """Returns json representation of JumpStartAlternativeConfig object."""
+        json_obj = {att: getattr(self, att) for att in self.__slots__ if hasattr(self, att)}
+        return json_obj
+
+    @property
+    def resolved_config(self) -> Dict[str, Any]:
```

Comment:
this logic won't work for dicts inside dicts. can you instead do:
```

def deep_override_dict(dict1: dict, dict2: dict) -> dict:
    """Overrides any overlapping contents of dict1 with the contents of dict2."""
    flattened_dict = flatten(dict1)
    flattened_dict.update(flatten(dict2))
    return unflatten(flattened_dict) if flattened_dict else {}
```

Reply:
I remembered I used something similar before to this but right now I don't see a native Python supported method `flatten`.  There's something similar but doesn't seem it trivially support `unflatten`: https://pandas.pydata.org/docs/reference/api/pandas.json_normalize.html